fsclient/client.go

```
@@ -0,0 +1,115 @@
+package fsclient
+
+import (
+	"bytes"
+	"io"
+	"io/ioutil"
+	"log"
+	"mime/multipart"
+	"net/http"
+	url2 "net/url"
+)
+
+type FileServerClient struct {
+	adr string
+	username string
+	password string
+}
+
+func NewFileServerClient(adr string, username string, password string) *FileServerClient{
+	return &FileServerClient{
+		adr: adr,
+		username: username,
+		password: password,
+	}
+}
+
+func (fsc *FileServerClient) UploadFile (url string, reader *io.Reader, filename string) error {
+	fullURL,err := url2.Parse(fsc.adr)
+	if err != nil {
+		return err
+	}
+	fullURL.Path = url
+	url = fullURL.String()
+
+	body := &bytes.Buffer{}
+	writer := multipart.NewWriter(body)
+	part, err := writer.CreateFormFile("file", filename)
+
+	if err != nil {
+		log.Fatal(err)
+		return err
+	}
+
+	_,err = io.Copy(part, *reader)
+	if err != nil {
+		log.Fatal(err)
+		return err
+	}
+	err = writer.Close()
+	if err != nil {
+		log.Fatal(err)
+		return err
+	}
+	request, err := http.NewRequest("POST", url, body)
+	request.SetBasicAuth(fsc.username, fsc.password)
+
+	if err != nil {
+		log.Fatal(err)
+		return err
+	}
+
+	request.Header.Add("Content-Type", writer.FormDataContentType())
+	client := &http.Client{}
+
+	response, err := client.Do(request)
+
+	if err != nil {
+		log.Fatal(err)
+		return err
+	}
+	defer response.Body.Close()
```

don't ignore the error here